azalea-client/src/mining.rs

```diff
@@ -66,6 +76,66 @@ impl Client {
             position,
         });
     }
+
+    /// When enabled, the bot will mine any block that it is looking at if it is reachable.
+    /// By default, reachability is set to 5 blocks.
+    pub fn auto_mine(&self, enabled: bool) {
+        let mut ecs = self.ecs.lock();
+        let mut entity_mut = ecs.entity_mut(self.entity);
+
+        if enabled {
+            entity_mut.insert(AutoMine);
+        } else {
+            entity_mut.remove::<AutoMine>();
+        }
+    }
+}
+
+#[derive(Component)]
+pub struct AutoMine;
```

Comment:
Tbh I'm not a fan of this name since to me it sounds like it would automatically mine like Baritone. I was thinking like LeftClickHeld or something but that might be too verbose. Feel free to suggest better names, please :)
(maybe look at the decompiled Minecraft source for naming ideas, I do that sometimes)

Reply:
I was thinking of calling it `LeftClickMine` as it is a implementation of holding left click but only for mining, it can't attack entities so `LeftClickHeld` might be little misleading. Other name i came up with was `TriggerLeftClick` or `TriggerLeftClickMine` but this might be too verbose. What do you think?

Reply:
Sure, `LeftClickMine` sounds fine.